samples/snippets/src/main/java/com/example/datastore/filters/OrFilterQuery.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.example.datastore.filters;
+
+// sample-metadata:
+//   title: Create a union between two filters
+//   description: Create a union between two filters (logical OR operator)
+
+// [START datastore_query_filter_or]
+import com.google.cloud.datastore.Datastore;
+import com.google.cloud.datastore.DatastoreOptions;
+import com.google.cloud.datastore.Entity;
+import com.google.cloud.datastore.Query;
+import com.google.cloud.datastore.QueryResults;
+import com.google.cloud.datastore.StructuredQuery.CompositeFilter;
+import com.google.cloud.datastore.StructuredQuery.Filter;
+import com.google.cloud.datastore.StructuredQuery.PropertyFilter;
+
+public class OrFilterQuery {
+  public static void invoke() {
+    
+    // Instantiates a client
+    Datastore datastore = DatastoreOptions.getDefaultInstance().getService();
+    String propertyName = "description";
+
+    // Create the two filters
+    Filter orFilter = CompositeFilter.or(
+        PropertyFilter.eq(propertyName, "Feed cats"),
+        PropertyFilter.eq(propertyName, "Buy milk")
+      );
+
+    // Build the query
+    Query<Entity> query = Query.newEntityQueryBuilder()
+        .setKind("Task")
+        .setFilter(orFilter)
+        .build();
+
+    // Get the results back from Datastore
+    QueryResults<Entity> results = datastore.run(query);
+
+    if (results.hasNext()) {
```

Comment:
maybe throw an exception in the else since we always expect something to be returned here

Reply:
Changed. I had intended to put `while` there but accidentally put `if`. Nice catch!

*Note*: I'm assuming that this is the correct way to iterate through a list of Datastore entities in Java. Please let me know if there is a better way.